src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs

```diff
@@ -187,13 +188,13 @@ private static string GetFixCategory(DiagnosticSeverity severity)
                 // This means that when we're called from the UI thread, we never try to go back to the
                 // UI thread.
                 TextSpan? selection = null;
-                if (IsForeground())
+                if (_threadingContext.JoinableTaskContext.IsOnMainThread)
```

Comment:
This block is really strange. A simple `SwitchToMainThreadAsync` (which returns immediately if already on the main thread) would have been sufficient.

Reply:
Sure. I'll file followup. I don't want to change anything here

Reply:
https://github.com/dotnet/roslyn/issues/73048